diff --git a/lib/factory-name.js b/lib/factory-name.js
index 27cf46b94809cb1a161a4da2bfa221137c95c9d5..72c988d8eeee8d651b3b58215a5fd85358fed8ac 100644
--- a/lib/factory-name.js
+++ b/lib/factory-name.js
@@ -26,11 +26,11 @@ export function factoryName(effects, ok, nok, type) {
 
   /** @type {State} */
   function name(code) {
-    if (code === 45 || code === 95 || asciiAlphanumeric(code)) {
+    if (code && code !== 10 && code > 0) {
       effects.consume(code);
       return name;
     }
     effects.exit(type);
-    return self.previous === 45 || self.previous === 95 ? nok(code) : ok(code);
+    return self.previous === 45 || self.previous === 95 || self.previous === 32 ? nok(code) : ok(code)
   }
 }
diff --git a/package.json b/package.json
index 2725b66483973753c473d47783bfb62d94a251a2..6dd541898b97e7db370004e1d0f522f8b35b3333 100644
--- a/package.json
+++ b/package.json
@@ -26,7 +26,6 @@
   "sideEffects": false,
   "type": "module",
   "exports": {
-    "development": "./dev/index.js",
     "default": "./index.js"
   },
   "files": [
