diff --git a/lib/factory-name.js b/lib/factory-name.js
index 0d6ccf284f019cb758a0b64fb5e1d25a39a3c5e7..1d5c3bb1128c289169b07f08d46933592c26bab6 100644
--- a/lib/factory-name.js
+++ b/lib/factory-name.js
@@ -26,9 +26,9 @@ export function factoryName(effects, ok, nok, type) {
 
   /** @type {State} */
   function name(code) {
-    if (code === null || markdownLineEnding(code) || unicodeWhitespace(code) || unicodePunctuation(code) && code !== 45 && code !== 95) {
+    if (code === null || markdownLineEnding(code) || unicodePunctuation(code) && code !== 45 && code !== 95) {
       effects.exit(type);
-      return self.previous === 45 || self.previous === 95 ? nok(code) : ok(code);
+      return self.previous === 45 || self.previous === 95 || self.previous === 32 ? nok(code) : ok(code);
     }
     effects.consume(code);
     return name;
diff --git a/package.json b/package.json
index 68a831665872d19b9a37e16b65e9a11f388457ae..65fdbccf32093d12a2d9b64050a9d49f80c46716 100644
--- a/package.json
+++ b/package.json
@@ -28,7 +28,6 @@
     "xo": "^0.60.0"
   },
   "exports": {
-    "development": "./dev/index.js",
     "default": "./index.js"
   },
   "files": [
