diff --git a/dist/lib/renderers/utils.js b/dist/lib/renderers/utils.js
index 971b51e38cf5370ec88e9e11b481659fb92eb76d..bc98555d82d1f9d917e830b710693895637f7d89 100644
--- a/dist/lib/renderers/utils.js
+++ b/dist/lib/renderers/utils.js
@@ -10,7 +10,7 @@ export const encodeInput = (input)=>{
         throw new Error('Input should not start with `/`');
     }
     if (input.endsWith('/')) {
-        throw new Error('Input should not end with `/`');
+      input = input.slice(0, -1);
     }
     return input + '.txt';
 };
diff --git a/dist/router/common.js b/dist/router/common.js
index 5882d7aba4ea815427c20810c101154ec4fd938a..01b19680935982a25c456f0b3a7db939f9ef2b39 100644
--- a/dist/router/common.js
+++ b/dist/router/common.js
@@ -1,5 +1,6 @@
 export function getComponentIds(path) {
-    const pathItems = path.split('/').filter(Boolean);
+  const decodedPath = decodeURIComponent(path);
+    const pathItems = decodedPath.split('/').filter(Boolean);
     const idSet = new Set();
     for(let index = 0; index <= pathItems.length; ++index){
         const id = [
diff --git a/dist/router/defineRouter.js b/dist/router/defineRouter.js
index d09f9ed9290070b61ce0e49a460be25ccbf96ee3..9e09333dd7c8a9c688bcf2bc3b78454ca7c035b7 100644
--- a/dist/router/defineRouter.js
+++ b/dist/router/defineRouter.js
@@ -18,7 +18,8 @@ export function defineRouter(getPathConfig, getComponent) {
     const has404Promise = pathConfigPromise.then((pathConfig)=>pathConfig.some(({ is404 })=>is404));
     const existsPath = async (pathname)=>{
         const pathConfig = await pathConfigPromise;
-        return pathConfig.some(({ path: pathSpec })=>getPathMapping(pathSpec, pathname));
+        const decodedPath = decodeURIComponent(pathname);
+        return pathConfig.some(({ path: pathSpec })=>getPathMapping(pathSpec, decodedPath));
     };
     const shouldSkip = {};
     const renderEntries = async (input, searchParams)=>{
